src/Integration.Vsix/Analysis/SingleFileMonitor.cs

```diff
@@ -131,18 +151,11 @@ private void OnFileChanged(object sender, System.IO.FileSystemEventArgs args)
             {
                 fileWatcher.EnableRaisingEvents = false;
 
-                // We're trying to ignore duplicate events by checking the last-write time.
-                // However, the precision of DateTime means that it is possible for separate events
-                // that happen very close together to report the same last-write time. If that happens,
-                // we'll be ignoring a "real" notification.
-                var currentTime = File.GetLastWriteTimeUtc(args.FullPath);
-                if (args.ChangeType != WatcherChangeTypes.Renamed && currentTime == lastWriteTime)
```

Comment:
Looking at `ChangeType`, it seems to be a Flags enum, which seems to suggest that the file system can aggregate multiple events.
https://learn.microsoft.com/en-us/dotnet/api/system.io.watcherchangetypes?view=net-6.0

The previous check was for `ChangeType != WatcherChangeTypes.Renamed`, whereas the new is for `ChangeType == WatcherChangeTypes.Changed`. I'm not sure either of them is the correct check, we should've been using HasFlag, although I'm not sure why we are ignoring duplicates based on change type. @duncanp-sonar do you remember what was the logic behind this?

Reply:
@rita-sonarsource no, I don't remember what the logic was, unfortunately. In theory, any kind of change (creation, deletion, rename, or change) could have an impact on the set of rules to be run.